File: api/src/main/java/com/ungalthedal/api/model/RegisterRequest.java
package com.ungalthedal.api.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class RegisterRequest {
    private String userName;
    private String firstName;
    private String lastName;
    private String gender;
    private String mobile;
    private String email;
    private String password;
}

File: ui/src/app/core/models/auth/RegisterRequest.ts
import {FormGroup} from "@angular/forms";

export class RegisterRequest {
  firstName: string = '';
  lastName: string = '';
  gender: string = '';
  mobile: string = '';
  email: string = '';
  password: string = '';

  copy(that: FormGroup): RegisterRequest {
    this.firstName = that.controls['firstName'].value
    this.lastName = that.controls['lastName'].value
    this.gender = that.controls['gender'].value
    this.mobile = that.controls['countryCde'].value + that.controls['mobile'].value
    this.email = that.controls['email'].value
    this.password = that.controls['password'].value
    return this
  }
}
